### PR TITLE
Move CI from TravisCI to GitHub Actions

### PR DESCRIPTION
**Pull request by jotompki**
Want to slowly migrate away from travis CI. testing this as a proof of concept to see how portable our workflows are. Going to keep travisCI until we are confident in this and can port the release workflow as well.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/pr-ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8]
```
Comment (PatMyron) [inline .github/workflows/pr-ci.yml:19]: Should we [start testing](https://github.com/actions/python-versions/blob/ac42875cb8afbc2ce400bf6db60105d6ea4c50f0/versions-manifest.json#L95) for [Python 3.9](https://docs.python.org/release/3.9.0/whatsnew/3.9.html) as well?
```suggestion
        python: [3.6, 3.7, 3.8, 3.9]
```

**Comment by jotompki** [comment]
Python 3.9 fails with these errors from MyPI:
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

src/cloudformation_cli_python_lib/interface.py:12: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:61: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:75: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:80: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:81: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:90: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:92: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:94: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:95: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:96: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:98: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:122: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:131: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:132: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:133: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:134: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:135: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:136: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:137: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:138: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:139: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:140: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:141: error: syntax error in type comment
src/cloudformation_cli_python_lib/interface.py:142: error: syntax error in type comment
```

Going to remove and file an issue to address whenever we want to support python3.9

File: .github/workflows/pr-ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+    runs-on: ubuntu-latest
```
Comment (ammokhov) [inline .github/workflows/pr-ci.yml:16]: curious if we should use centos?

**Review by ammokhov — APPROVED** [review]

File: .github/workflows/pr-ci.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install --upgrade pip mypy 'attrs==19.2.0' -r https://raw.githubusercontent.com/aws-cloudformation/aws-cloudformation-rpdk/master/requirements.txt
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (PatMyron) [inline .github/workflows/pr-ci.yml:30]:
```suggestion
        pip install --upgrade mypy 'attrs==19.2.0' -r https://raw.githubusercontent.com/aws-cloudformation/aws-cloudformation-rpdk/master/requirements.txt
```

**Review by PatMyron — APPROVED** [review]

File: .github/workflows/pr-ci.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
```
Comment (PatMyron) [inline .github/workflows/pr-ci.yml:15]: Is this environment variable used for anything btw?
```suggestion
```

File: .github/workflows/pr-ci.yml (hunk 2 above)
Comment (PatMyron) [reply .github/workflows/pr-ci.yml:16]: we should test on [`macos-latest` and `windows-latest`](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources) as well though

**Comment by PatMyron** [comment]
Could get rid of the Travis badge:

https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/ae8533f5a375883023969e9c19436c042d8721df/README.md#L13

**Pull request merged**